domain/model/state/state.go

```diff
@@ -1111,3 +1120,37 @@ AND cloud_uuid = ?
 	}
 	return nil
 }
+
+// addOwnerPermissions inserts an Admin permission for the owner on the given
+// model during create model. The caller has already validated the user exists.
+func addOwnerPermissions(ctx context.Context, tx *sql.Tx, modelUUID, ownerUUID string) error {
+	permUUID, err := internaluuid.NewUUID()
+	if err != nil {
+		return err
+	}
+
+	permStmt := `
+INSERT INTO permission (uuid, access_type_id, object_type_id, grant_to, grant_on)
```

Comment:
I know you're doing this in an exiting standard transaction, but there is a lot in this state particularly suited to SQLair usage. We can talk about it in Madrid.